sdk/KoreBuild/modules/frameworkbuild/3-compile.targets

```diff
@@ -0,0 +1,75 @@
+<Project>
+  <PropertyGroup>
+  	<CompileDependsOn>$(CompileDependsOn);CompileProjects;CompileRuntimeStore</CompileDependsOn>
+  </PropertyGroup>
+
+  <Target Name="CompileProjects" DependsOnTargets="PrepareProjects">
+    <KoreBuild
+      Projects="@(KoreBuildProject)"
+      Targets="Package"
+      BuildInParallel="true"
+      Properties="$(RepoBuildProps)" />
+
+    <PropertyGroup>
+      <!-- compilation done! Skip in future invocations -->
+      <RepoBuildProps>$(RepoBuildProps);NoBuild=true</RepoBuildProps>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <SymbolsPackageArtifactFile Include="%(KoreBuildProject.RootDir)artifacts\build\*.symbols.nupkg" />
+      <PackageArtifactFile Include="%(KoreBuildProject.RootDir)artifacts\build\*.nupkg" Exclude="@(SymbolsPackageArtifactFile)" />
+      <AdditionalArtifactFiles Include="%(KoreBuildProject.RootDir)artifacts\**\*" Exclude="@(PackageArtifactFile)" Repository="%(KoreBuildProject.Name)" />
+    </ItemGroup>
+
+    <!-- TODO do we still need this? -->
+    <Copy SourceFiles="@(PackageArtifactFile)" DestinationFolder="$(BuildDir)" />
+    <Copy SourceFiles="@(SymbolsPackageArtifactFile)" DestinationFolder="$(ArtifactsDir)symbols\" />
+
+    <Copy
+      SourceFiles="%(AdditionalArtifactFiles.Identity)"
+      DestinationFolder="$(ArtifactsDir)%(Repository)\%(RecursiveDir)%(FileName)%(Extension)" />
+
+    <!-- TODO implement task. Reads a nupkg file and creates and item with metadata about its id, version, and file location -->
+    <GetPackageIdentity Files="@(PackageArtifactFile)">
+
+      <Output TaskParameter="PackageIdentity" ItemName="PackageReference" />
+    </GetPackageIdentity>
+
+    <!-- TODO maybe we can make this more generic? e.g. just assign a Category metadata? -->
+    <!-- TODO implement task -->
+    <!-- Categories packages -->
+    <ClassifyPackages
+      CsvFile="$(RepositoryRoot)packages.csv"
```

Comment:
Should we keep packages.csv or use another format? i.e. MSBuild items

Reply:
packages.csv for now I'd assume. There's parsing logic in coherence that assumes that format. Unless we want to change the format there now as well. https://github.com/aspnet/Coherence/blob/f7fcc555ffa897ef1c115f3d2c9fd4220752e6be/build/tasks/CopyPackagesToSplitFolders.cs

Reply:
Oops that's the dev version. I think rel/2.0.0 is here: https://github.com/aspnet/BuildTools/tree/rel/2.0.0/src/SplitPackages

Reply:
Let's copy the CopyPackagesToSplitFolders.cs task. SplitPackages did a whole bunch of other stuff that it didn't need to.

Reply:
Will also need to update the CopyPackagesToSplitFolders.cs task. It currently parses the csv but it should be trivial to take an ItemGroup as input instead and operate on the individual metadata.

Reply:
starting to seem like the packages.csv is more effort than its worth. Maybe we should make these an ItemGroup with well-known metadata. i.e.

```xml
<ItemGroup>
  <ICanHazPackage Include="Microsoft.AspNetCore" OutputFolder="ship" Metapackage="true" />
</ItemGroup>
```

Reply:
Yup that's the idea I was getting to in https://github.com/aspnet/BuildTools/pull/401#discussion_r138785937. The metadata names here are better though.

Reply:
No clue what to name all these groups of packages though.

Reply:
ShipPackage, NoShipPackages, ShipOOBPackages, RuntimeStorePackages. These should be sufficient for now.